thalerj/openbmctool.py

```diff
@@ -0,0 +1,1490 @@
+        return res.text
+    elif(args.identcmd == 'off'):
+        print("Attempting to turn identify light off...:")
+        url="https://"+host+"/xyz/openbmc_project/led/groups/enclosure_identify/attr/Asserted"
+        httpHeader = {'Content-Type':'application/json'}
+        data = '{"data":false}'
+        res = session.put(url, headers=httpHeader, data=data, verify=False, timeout=20)
+        return res.text
+    elif(args.identcmd == 'status'):
+        url="https://"+host+"/xyz/openbmc_project/led/groups/enclosure_identify"
+        httpHeader = {'Content-Type':'application/json'}
+#         print(url)
+        res = session.get(url, headers=httpHeader, verify=False, timeout=20)
+        status = json.loads(res.text)['data']
+        if(args.json):
+            return status
+        else:
+            if status['Asserted'] == 0:
+                return "Identify light is off"
+            else:
+                return "Identify light is blinking"
+    else:
+        return "Invalid chassis identify command"
+
+"""
+     controls the different chassis commands
+       
+     @param host: string, the hostname or IP address of the bmc
+     @param args: contains additional arguments used by the fru sub command
+     @param session: the active session to use
+     @param args.json: boolean, if this flag is set to true, the output will be provided in json format for programmatic consumption 
+""" 
+def chassis(host, args, session):
+    if(hasattr(args, 'powcmd')):
+        result = chassisPower(host,args,session)
+    elif(hasattr(args, 'identcmd')):
+        result = chassisIdent(host, args, session)
+    else:
+        return "to be completed"
+    return result
+"""
+     Downloads a dump file from the bmc
+       
+     @param host: string, the hostname or IP address of the bmc
+     @param args: contains additional arguments used by the collectServiceData sub command
+     @param session: the active session to use
+     @param args.json: boolean, if this flag is set to true, the output will be provided in json format for programmatic consumption 
+"""
+def bmcDumpRetrieve(host, args, session):
+    httpHeader = {'Content-Type':'application/json'}
+    dumpNum = args.dumpNum
+    if (args.dumpSaveLoc is not None):
+        saveLoc = args.dumpSaveLoc
+    else:
+        saveLoc = '/tmp'
+    url ='https://'+host+'/download/dump/' + str(dumpNum)
+    try:
+        r = session.get(url, headers=httpHeader, stream=True, verify=False, timeout=20)
+        if (args.dumpSaveLoc is not None):
+            if os.path.exists(saveLoc):
+                if saveLoc[-1] != os.path.sep:
+                    saveLoc = saveLoc + os.path.sep
+                filename = saveLoc + host+'-dump' + str(dumpNum) + '.tar.xz'
+                
+            else:
+                return 'Invalid save location specified'
+        else:
+            filename = '/tmp/' + host+'-dump' + str(dumpNum) + '.tar.xz'
```

Comment:
As mentioned elsewhere, please don't do this. Either print the a name generated with `os.mkstemp()` or ask the user to specify an output file